DateTimeCalendar/App.java

```diff
@@ -0,0 +1,605 @@
+            // System.out.println(checkDiff.getDate().getDayspan());
+
+        }
+        // System.out.println(dateTimeDiffControlObj.getDate().getDayspan());
+
+        // DateTime checkDiff = dateTimeDiffControlObj.diff(diffBy[0]);
+
+        // System.out.println(String.format("D:, H:, M:, S:%d", checkDiff.getTime().getSecond(false) ));
+    }
+
+    private static void unitTestCalendarCalcDayOfWeek(ShowTestOutput selectedTypeOfOutput){
+      //1.) Provide a date
+      //2.) Give an expected int day of the week value
+      //3.) Check for match
+
+      Date[] inputValues = new Date[10];
+      inputValues[0] = new Date(1900, Date.Month.JAN, 1);
+      inputValues[1] = new Date(1900, Date.Month.JAN, 2);
+      inputValues[2] = new Date(1900, Date.Month.FEB, 1);
+      inputValues[3] = new Date(1900, Date.Month.MAR, 1);
+      inputValues[4] = new Date(1900, Date.Month.DEC, 30);
+      inputValues[5] = new Date(1904, Date.Month.JAN, 1);
+      inputValues[6] = new Date(1904, Date.Month.JAN, 2);
+      inputValues[7] = new Date(1904, Date.Month.FEB, 1);
+      inputValues[8] = new Date(1904, Date.Month.MAR, 1);
+      inputValues[9] = new Date(1904, Date.Month.DEC, 30);
+
+      int[] expectedValue = new int[10];
+      expectedValue[0] = 1;
+      expectedValue[1] = 2;
+      expectedValue[2] = 4;
+      expectedValue[3] = 4;
+      expectedValue[4] = 0;
+      expectedValue[5] = 5;
+      expectedValue[6] = 6;
+      expectedValue[7] = 1;
+      expectedValue[8] = 2;
+      expectedValue[9] = 5;
+
+      for (int i = 0; i < inputValues.length; i++) {
+        boolean testPass = false;
+        int payload = Calendar.calcDayOfWeek(inputValues[i]);
+        //testing print statement
+        Calendar.calcWeeksInMonth(inputValues[i]);
+
+        if (expectedValue[i] == payload) {
+          testPass = true;
+        }
+
+        if (selectedTypeOfOutput == ShowTestOutput.ALL) {
+          System.out.println(String.format("Test Calendar.calcDayOfWeek() inputValue:%s// EO:%d AO:%d // TestPass:%s", inputValues[i].toString(), expectedValue[i], payload, testPass));
+        } else if (selectedTypeOfOutput == ShowTestOutput.ON_ERROR && !testPass) {
+          System.out.println(String.format("Test Calendar.calcDayOfWeek() inputValue:%s// EO:%d AO:%d // TestPass:%s", inputValues[i].toString(), expectedValue[i], payload, testPass));          
+        } 
+      }
+    }
+
+    private static void genericTestMethod(Object[] input, Object expectedOutput) {
+        //I want to test <(input), (expectedOutput), (class to test in), (method to use)>
+        //current thinking: since I can find the class that is input, should I have an override for equals in each of my classes to make testing easier
+        System.out.println(input[0].toString());
+        System.out.println(input[0].getClass());
+        System.out.println(input[1].hashCode());
+    }
+
+    private static class DateTestCase {
```

Comment:
Why static?

Reply:
Simple answer: I made it a class before, and the compiler complained, saying that the method wasn't static and wouldn't compile. So I made it static.